model/task_queue.go

```diff
@@ -81,20 +82,24 @@ func (self *TaskQueue) Save() error {
 }
 
 func (self *TaskQueue) FindTask(spec TaskSpec) *TaskQueueItem {
+	if spec.Group == "" {
+		return nil
+	}
+
 	for _, it := range self.Queue {
-		if spec.ProjectID != "" && it.Project != spec.ProjectID {
+		if spec.ProjectID == "" || it.Project != spec.ProjectID {
```

Comment:
shouldn't the `spec.ProjectID` case return early, above the for loop? (same with `spec.Version == ""`, `specBuildVariant == ""`)